plugins/modules/ipaconfig.py

```diff
@@ -0,0 +1,441 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Authors:
+#   Chris Procter <cprocter@redhat.com>
+#
+# Copyright (C) 2019 Red Hat
+# see file 'COPYING' for use and warranty information
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.0",
+    "supported_by": "community",
+    "status": ["preview"],
+}
+
+
+DOCUMENTATION = '''
+---
+module: ipa_config
+author: chris procter
+short_description: Modify IPA global config options
+description:
+- Modify IPA global config options
+options:
+    ipaadmin_principal:
+        description: The admin principal
+        default: admin
+    ipaadmin_password:
+        description: The admin password
+        required: false
+    ipamaxusernamelength:
+        description: Set the maximum username length between 1-255
+        required: false
+        aliases: ["maxusername"]
+    ipahomesrootdir:
+        description: Set the default location of home directories
+        required: false
+        aliases: ["homedirectory"]
+    ipadefaultloginshell:
+        description: Set the default shell for new users
+        required: false
+        aliases: ["defaultshell"]
+    ipadefaultprimarygroup:
+        description: Set the default group for new users
+        required: false
+        aliases: ["defaultgroup"]
```

Comment:
The alias name should be the variable name. You don't need to use the 'ipa*' name.